test/system/src/test/java/io/pravega/test/system/SegmentReaderAPITest.java

```diff
@@ -0,0 +1,483 @@
+        ControllerImpl controller = new ControllerImpl(ControllerImplConfig.builder().clientConfig(clientConfig).build(),
+                connectionFactory.getInternalExecutor());
+
+        StreamConfiguration streamConfiguration = StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.fixed(1))
+                .build();
+        Stream stream = Stream.of(streamScope, streamName);
+
+        @Cleanup
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(streamScope, controller, connectionFactory);
+        log.info("Invoking Writer test with Controller URI: {}", controllerURI);
+
+        StreamManager streamManager = StreamManager.create(clientConfig);
+        assertTrue("Creating Scope", streamManager.createScope(streamScope));
+        assertTrue("Creating stream", streamManager.createStream(streamScope, streamName, streamConfiguration));
+
+        @Cleanup
+        BatchClientFactory batchClient = BatchClientFactory.withScope(streamScope, clientConfig);
+
+        StreamCut streamCut0 = streamManager.fetchStreamInfo(streamScope, streamName).join().getHeadStreamCut();
+        log.info("Initial stream streamCut0 {}", streamCut0);
+        assertEquals(1, streamCut0.asImpl().getPositions().size());
+
+        List<Segment> list = streamCut0.asImpl().getPositions().keySet().stream().collect(Collectors.toList());
+
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName, new UTF8StringSerializer(),
+                EventWriterConfig.builder().build());
+
+        // write events to stream 30 *10  = 300 bytes
+        writeEvents(10, writer);
+        writer.flush();
+
+        //Requested next stream cut at a distance of 170 bytes, and getting the next approx offset as a response.
+        StreamCut streamCut1 = batchClient.getNextStreamCut(streamCut0, 170L);
+        long streamCut1Position = streamCut1.asImpl().getPositions().get(list.get(0)).longValue();
+        log.info("Next stream cut1 {} streamCut1 position {}", streamCut1, streamCut1Position);
+        assertTrue(180 <= streamCut1Position);
+
+        @Cleanup
+        ReaderGroupManager groupManager = ReaderGroupManager.withScope(streamScope, controllerURI);
+        ReaderGroupConfig readerGroupConfig1 = getReaderGroupConfig(streamCut0, streamCut1, stream);
+        groupManager.createReaderGroup(readerGroupName, readerGroupConfig1);
+
+        @Cleanup
+        ReaderGroup readerGroup = groupManager.getReaderGroup(readerGroupName);
+
+        // Reading the events between the two streamCut, startStreamCut = streamCut0 and endStreamCut = streamCut1
+        @Cleanup
+        EventStreamReader<String> reader0 = clientFactory.createReader(readerName,
+                readerGroupName,
+                new UTF8StringSerializer(),
+                ReaderConfig.builder().build());
+
+        int readCount1 = readEvent(reader0, streamCut1Position / 30);
+        reader0.close();
```

Comment:
Why we need to close reader every time? Can we proceed with same reader to read next event?

Reply:
We cannot proceed with the same reader, because after resetting the reader group, older readers does not exist.
Due to that before resetting the reader group, closing it and after resetting the reader group creating the new reader to read the events between the stream cut.